docs/pages/result.rst

```diff
@@ -43,6 +43,42 @@ Make sure to check out how to compose container with
 ``flow`` or :ref:`pipe`!
 Read more about them if you want to compose your containers easily.
 
+Pattern Matching
+----------------
+
+``Result`` values can be matched using the new feature of Python 3.10,
+`Structural Pattern Matching <https://www.python.org/dev/peps/pep-0622/>`_,
+see the example below:
+
+.. code:: python
+
+  from returns.result import Success, Failure, safe
+
+  @safe
+  def div(first_number: int, second_number: int) -> int:
+      return first_number // second_number
+
+  match div(1, 0):
+      # Matches if the result stored inside `Success` is `10`
+      case Success(10):
+          print('Result is "10"')
+
+      # Same as above but using match-by-name
+      case Success(inner_value=20):
```

Comment:
Are you sure that this will work? `_inner_value` is protected.

Reply:
Can we somehow make this a real doctest? And skip it if it's not a `python3.10` environment: https://stackoverflow.com/questions/42708809/python-doctest-skip-a-test-conditionally

Reply:
> Are you sure that this will work? _inner_value is protected.

Yep, I've tested it! How does it work? I don't know hahaha
Look at here: https://www.python.org/dev/peps/pep-0622/#special-attribute-match-args

```text
...
In deciding what names should be available for matching, the recommended practice is that class patterns should be the mirror of construction; that is, the set of available names and their types should resemble the arguments to __init__().
...
```

Reply:
What about doctests?

Reply:
We should do that!! I'm just worried about how to skip, following the link you sent we'll need to put `# doctest: +PYTHON_310` in every single line!

Reply:
In this case, we can store this file in https://github.com/dry-python/returns/tree/master/tests/test_examples/test_pattern_patching and ignore this folder in `collect` hook of `pytest` https://docs.pytest.org/en/6.2.x/reference.html#collection-hooks

Reply:
I'll do it tonight!